### ISSUE TITLE
Remove jQuery

### ISSUE THREAD

**Issue by pushcx** [lobsters/lobsters#554]
In the time the site's been around, cross-browser js functionality and compatibility has improved to the point that we can drop jQuery as a dependency.

We only have ~600 lines of JavaScript and none of it's very complex. We don't serve JS to logged-out users, but for logged-in users it adds up to 140 KB (45KB compressed) of data (standard Rails asset caching, so caches very well). Removing jQuery is not a huge win, so this is pretty far down on my to-do list. Still, it's come up in chat a couple times in the last few months and I thought I'd post it for anyone who might like the exercise of replacing jQuery with no-framework js.

We have two libraries in use. Autosize resizes textareas rather than scroll them and doesn't depend on jQuery - so there's no changes here. Select2 is used for filtering and tab-completing the tag dropdown when creating and editing story changes. It depends on jQuery and will need to be [rewritten](https://github.com/select2/select2/issues/3652) or replaced. (Ideally with identical behavior, because we haven't touched it in six years and submitters have muscle memory for it.)

**Comment by daau** [comment]
[Choices](https://github.com/jshjohnson/Choices) seems like a good Select2 alternative which doesn't depend on jQuery.

**Comment by pushcx** [comment]
From their demo, Choices looks like a good fit, thanks for the suggestion. We'd have to add a hook so that tab completes the currently-selected choice, but that's probably not too bad.

**Comment by daau** [comment]
Great! I'll submit a PR shortly with this feature.

**Comment by Carpetsmoker** [comment]
I did some work on s/select2/choices/: https://github.com/Carpetsmoker/lobsters/tree/choices

Can't say I'm too impressed with it. The API is annoying. The documentation isn't very well written, and several things are non-obvious how to do (like not have it muck about with the width of the input), it's a lot larger than select2 (56k vs. 213k), and I had to heavily modify the shipped CSS to make it vaguely behave.

I see if I can find something better later, or maybe I'll write my own simple function. It's not a very hard problem and probably would have finished it by now instead of making that Choices thing behave.

**Comment by drakmail** [comment]
How about https://github.com/yairEO/tagify ?

**Comment by Carpetsmoker** [comment]
Thanks. I spent some time with that as well, and eventually just gave up.

I did some other work using "pure" JavaScript a while ago, and the experience wasn't a happy one. The DOM API offends a lot of my sensibilities on how a good API should work, which is basically just a nice way of saying that I think it's pretty terrible. `elem.parentNode.removeChild(elem)`, seriously?

And what's the value? Save maybe 10k on the compressed JS file? Meh. Seems like a poor ROI. I thought it would be easy with a few `:%s/.../.../` but it's not really, and IMHO it will make the code worse in a way.

In short, not really worth my time, and probably not really worth anyone's time IMHO.

**Comment by shelches** [comment]
@pushcx  Would you consider a straightforward list of checkboxes, tags, and descriptions that uses no JavaScript?

**Comment by pushcx** [comment]
As a fallback for no-js? Sure. But I'd like to preserve our existing, familiar behavior because it's worked very well for us.

**Comment by pushcx** [comment]
This was completed by @ymeynot45 in the linked PRs.
